### ISSUE TITLE
Looping over elements children

### ISSUE THREAD

**Issue by hir3npatel** [brython-dev/brython#479]
The docs mention that one should be able to do
(http://brython.info/static_doc/en/attributes.html)

```
for child in element:
    do_something_with_child
```

But in my experience clearing a div's children and re-populating it with new data, this does not work as expected:

```
for child in element:
    element.removeChild(child)
```

When run it seems to leave some of the elements children intact, and does not seem to clear them all out.

If I change it to this, then it works as expected (note the use of element.children):

```
for child in element.children:
  element.removeChild(child)
```

**Comment by kikocorreoso** [comment]
IMO there is something definitely broken. The actual API is not pretty clear and/or well documented and maybe it would be easier to just expose the JS API as it is well documented and there are tons of examples.

@PierreQuentel what do you think about it?

**Comment by PierreQuentel** [comment]
Thanks for the bug report.

I fixed it, but there are many things to review in DOM interface. Some things that are documented don't work (`del` for instance) and I added new methods (`select()` which is a little like jQuery `$()`, `closest()`) that are not yet documented.

It's not the best time for me to work on this, I'm on holidays, but it's something to do in the next weeks. The DOM API must be supported, but I think there should be more Pythonic ways to interact with the DOM, and some methods like element or mouse absolute position should remain available with less code than with the standard API.
